helper/builtinplugins/builtin.go

```diff
@@ -0,0 +1,30 @@
+package builtinplugins
+
+import (
+	"github.com/hashicorp/vault/plugins/database/mysql"
+	"github.com/hashicorp/vault/plugins/database/postgresql"
+)
+
+type BuiltinFactory func() (interface{}, error)
```

Comment:
Why does it need to return error? I was looking at the funcs, i.e. `New()`'s from the databases, that implements this (found from comments) and couldn't find a case where they would handle errors?